website/project/decorators.py

```diff
@@ -223,13 +220,12 @@ def wrapped(*args, **kwargs):
             #if not login user check if the key is valid or the other privilege
 
             kwargs['auth'].private_key = key
-            link_anon = None
             if not include_view_only_anon:
                 from osf.models import PrivateLink
                 try:
-                    link_anon = PrivateLink.find_one(Q('key', 'eq', key)).anonymous
-                except ModularOdmException:
-                    pass
+                    link_anon = PrivateLink.objects.get(key=key).anonymous
```

Comment:
Minor: Can be optimized slightly with `PrivateLink.objects.filter(key=key).values_list('anonymous', flat=True).get()`.